### PR TITLE
dm: fix panic of openapi operate-schema

### LINKED ISSUE #10839: DM worker panic when OpenAPI GET /api/v1/tasks/{task-name}/sources/{source-name}/migrate_targets
### What did you do?

curl -X 'GET'   'http://10.2.8.3:8561/api/v1/tasks/task-1/sources/mysql-01/migrate_targets'   -H 'accept: application/json'

### What did you expect to see?

_No response_

### What did you see instead?

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2d7c85d]

goroutine 223775 [running]:
github.com/pingcap/tidb/ddl/schematracker.(*InfoStore).AllSchemaNames(...)
        github.com/pingcap/tidb@v1.1.0-beta.0.20230412065101-c24a1cda6d26/ddl/schematracker/info_store.go:132
github.com/pingcap/tiflow/dm/pkg/schema.(*Tracker).AllSchemas(...)
        github.com/pingcap/tiflow/dm/pkg/schema/tracker.go:236
github.com/pingcap/tiflow/dm/syncer.(*Syncer).listMigrateTargets(0xc00207a900, 0xc002da1500)
        github.com/pingcap/tiflow/dm/syncer/schema.go:183 +0x5d
github.com/pingcap/tiflow/dm/syncer.(*Syncer).OperateSchema(0xc00207a900, {0x3faab90?, 0xc00056ed80}, 0xc002da1500)
        github.com/pingcap/tiflow/dm/syncer/schema.go:49 +0x185
github.com/pingcap/tiflow/dm/worker.(*SubTask).OperateSchema(0xc000d78460, {0x3faab90, 0xc00056ed80}, 0xc002da1500)
        github.com/pingcap/tiflow/dm/worker/subtask.go:678 +0x24c
github.com/pingcap/tiflow/dm/worker.(*SourceWorker).OperateSchema(0xc001346140, {0x3faab90, 0xc00056ed80}, 0xc002da1500)
        github.com/pingcap/tiflow/dm/worker/source_worker.go:1046 +0x1ae
github.com/pingcap/tiflow/dm/worker.(*Server).OperateSchema(0xc00169c120, {0x3faab90, 0xc00056ed80}, 0xc002da1500)
        github.com/pingcap/tiflow/dm/worker/server.go:848 +0x4a7
github.com/pingcap/tiflow/dm/pb._Worker_OperateSchema_Handler({0x35e11e0?, 0xc00169c120}, {0x3faab90, 0xc00056ed80}, 0xc0012df570, 0x0)
        github.com/pingcap/tiflow/dm/pb/dmworker.pb.go:3973 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001834000, {0x3fcc3a0, 0xc000c321a0}, 0xc001a0a360, 0xc000d7aa50, 0x63e9ad0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc001834000, {0x3fcc3a0, 0xc000c321a0}, 0xc001a0a360, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

### Versions of the cluster

v6.5.4

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #10839 

### What is changed and how it works?
- for a task in invalid stage(create task but no start), report an error instead of panic

#### Tests 

 - Integration test

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: dm/worker/subtask.go [hunk 1]
```diff
@@ -652,7 +652,7 @@ func (st *SubTask) Update(ctx context.Context, cfg *config.SubTaskConfig) error
 
 // OperateSchema operates schema for an upstream table.
 func (st *SubTask) OperateSchema(ctx context.Context, req *pb.OperateWorkerSchemaRequest) (schema string, err error) {
-	if st.Stage() != pb.Stage_Paused && req.Op != pb.SchemaOp_ListMigrateTargets {
+	if !(st.Stage() == pb.Stage_Paused || st.stage == pb.Stage_Running && req.Op == pb.SchemaOp_ListMigrateTargets) {
```
Comment (okJiang) [inline dm/worker/subtask.go:655]: new and use `ErrWorkerNotPausedAndRunningStage`?

Comment (GMHDBJD) [reply dm/worker/subtask.go:655]: no need, because the task didn't start yet

Comment (lance6716) [reply dm/worker/subtask.go:655]: it's hard to remember the priority of different boolean operators, can you refine the check? like 

switch req.Op {
...
}

**Review by lance6716 — APPROVED** [review]

File: dm/worker/subtask.go [hunk 2]
```diff
@@ -652,8 +652,15 @@ func (st *SubTask) Update(ctx context.Context, cfg *config.SubTaskConfig) error
 
 // OperateSchema operates schema for an upstream table.
 func (st *SubTask) OperateSchema(ctx context.Context, req *pb.OperateWorkerSchemaRequest) (schema string, err error) {
-	if st.Stage() != pb.Stage_Paused && req.Op != pb.SchemaOp_ListMigrateTargets {
-		return "", terror.ErrWorkerNotPausedStage.Generate(st.Stage().String())
+	switch req.Op {
+	case pb.SchemaOp_ListMigrateTargets:
+		if st.Stage() != pb.Stage_Running && st.Stage() != pb.Stage_Paused {
+			return "", terror.ErrWorkerNotPausedStage.Generate(st.Stage().String())
```
Comment (D3Hunter) [inline dm/worker/subtask.go:658]: not sure if we have a ErrInvalidState

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**